### ISSUE TITLE
Upgrade to 2.3.1 from 2.2.9 Resulting in Unstable Clusters (Networking Issues)

### ISSUE THREAD

**Issue by ekristen** [rancher/rancher#23723]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

Upgrade from 2.2.9 to 2.3.1

**Result:**

**Note:** The initial upgrade seemed to go ok. All clusters went unavailable while the cluster agents were updated and reconnected. It was only after everything appeared to be ok again that internal networking issues appeared. 

Unstable clusters.

This seems to be cross cutting between baremetal and cloud. Cloud plugin doesn't seem to have an affect.

Issues seem to be all networking related.

- All apps being served by ingress nginx resulted in 502 bad gateway. Logs seem to indicate DNS is resolving to 172.17.x.x ips instead of 10.43.x.x ips.
- DNS resolution outside of the cluster fails, unable to query google.com
- Cluster monitoring may stop working
- General i/o timeouts when attempting to make external connections

**Other details that may be helpful:**

All clusters were on 1.14.5

Basically I had to redeploy canal, coredns, and all my workloads to get networking and DNS to start working again. 

Editing of the cluster doesn't necessarily fix anything.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3.1
- Installation option (single install/HA): HA

**Comment by superseb** [comment]
This is the same as https://github.com/rancher/rancher/issues/23284, please provide any information that can help reproduce (more uncommon characteristics to the setup) in there as it doesn't reproduce 100%
